src/common/utils/UnhandledExceptionHandlerX64.h

```diff
@@ -0,0 +1,198 @@
+        return "EXCEPTION_ACCESS_VIOLATION";
+    case EXCEPTION_ARRAY_BOUNDS_EXCEEDED:
+        return "EXCEPTION_ARRAY_BOUNDS_EXCEEDED";
+    case EXCEPTION_BREAKPOINT:
+        return "EXCEPTION_BREAKPOINT";
+    case EXCEPTION_DATATYPE_MISALIGNMENT:
+        return "EXCEPTION_DATATYPE_MISALIGNMENT";
+    case EXCEPTION_FLT_DENORMAL_OPERAND:
+        return "EXCEPTION_FLT_DENORMAL_OPERAND";
+    case EXCEPTION_FLT_DIVIDE_BY_ZERO:
+        return "EXCEPTION_FLT_DIVIDE_BY_ZERO";
+    case EXCEPTION_FLT_INEXACT_RESULT:
+        return "EXCEPTION_FLT_INEXACT_RESULT";
+    case EXCEPTION_FLT_INVALID_OPERATION:
+        return "EXCEPTION_FLT_INVALID_OPERATION";
+    case EXCEPTION_FLT_OVERFLOW:
+        return "EXCEPTION_FLT_OVERFLOW";
+    case EXCEPTION_FLT_STACK_CHECK:
+        return "EXCEPTION_FLT_STACK_CHECK";
+    case EXCEPTION_FLT_UNDERFLOW:
+        return "EXCEPTION_FLT_UNDERFLOW";
+    case EXCEPTION_ILLEGAL_INSTRUCTION:
+        return "EXCEPTION_ILLEGAL_INSTRUCTION";
+    case EXCEPTION_IN_PAGE_ERROR:
+        return "EXCEPTION_IN_PAGE_ERROR";
+    case EXCEPTION_INT_DIVIDE_BY_ZERO:
+        return "EXCEPTION_INT_DIVIDE_BY_ZERO";
+    case EXCEPTION_INT_OVERFLOW:
+        return "EXCEPTION_INT_OVERFLOW";
+    case EXCEPTION_INVALID_DISPOSITION:
+        return "EXCEPTION_INVALID_DISPOSITION";
+    case EXCEPTION_NONCONTINUABLE_EXCEPTION:
+        return "EXCEPTION_NONCONTINUABLE_EXCEPTION";
+    case EXCEPTION_PRIV_INSTRUCTION:
+        return "EXCEPTION_PRIV_INSTRUCTION";
+    case EXCEPTION_SINGLE_STEP:
+        return "EXCEPTION_SINGLE_STEP";
+    case EXCEPTION_STACK_OVERFLOW:
+        return "EXCEPTION_STACK_OVERFLOW";
+    default:
+        return "UNKNOWN EXCEPTION";
+    }
+}
+
+/* Returns the index of the last backslash in the file path */
+int GetFilenameStart(CHAR* path)
+{
+    int pos = 0;
+    int found = 0;
+    if (path != NULL)
+    {
+        while (path[pos] != '\0' && pos < MAX_PATH)
+        {
+            if (path[pos] == '\\')
+            {
+                found = pos + 1;
+            }
+            ++pos;
+        }
+    }
+
+    return found;
+}
+
+void LogStackTrace()
+{
+    BOOL result;
+    HANDLE thread;
+    HANDLE process;
+    CONTEXT context;
+    STACKFRAME64 stack;
+    ULONG frame;
+    DWORD64 dw64Displacement;
+    DWORD dwDisplacement;
+
+    memset(&stack, 0, sizeof(STACKFRAME64));
+    memset(pSymbol, '\0', sizeof(*pSymbol) + MAX_PATH);
+    memset(&modulePath[0], '\0', sizeof(modulePath));
+    line.LineNumber = 0;
+
+    RtlCaptureContext(&context);
+    process = GetCurrentProcess();
+    thread = GetCurrentThread();
+    dw64Displacement = 0;
+    stack.AddrPC.Offset = context.Rip;
+    stack.AddrPC.Mode = AddrModeFlat;
+    stack.AddrStack.Offset = context.Rsp;
+    stack.AddrStack.Mode = AddrModeFlat;
+    stack.AddrFrame.Offset = context.Rbp;
+    stack.AddrFrame.Mode = AddrModeFlat;
+
+    std::stringstream ss;
+    for (frame = 0;; frame++)
+    {
+        result = StackWalk64(
```

Comment:
should we introduce error-checks for these api calls?

Reply:
Yes

Reply:
Added some handling. Note, that we have a global try-catch in the unhandled exception handler that will take care of unexpected exceptions.